### PR TITLE
docs: stricter bufname and bufnr types

### LINKED ISSUE #27444: Lua types for vim.fn.bufnr and vim.fn.bufname shouldn't be integer
### Problem

These actually take strings and integers

### Expected behavior

and so I think it should reflect that

### PR DESCRIPTION
**Pull request by MariaSolOs**
Closes #27444.

### CODE REVIEW

File: runtime/lua/vim/_meta/vimfn.lua [hunk 1]
```diff
@@ -599,7 +599,7 @@ function vim.fn.bufname(buf) end
 --- number necessarily exist, because ":bwipeout" may have removed
 --- them.  Use bufexists() to test for the existence of a buffer.
 ---
---- @param buf? any
+--- @param buf? number|string
```
Comment (wookayin) [inline runtime/lua/vim/_meta/vimfn.lua:602]: `integer|string`?

File: src/nvim/eval.lua [hunk 2]
```diff
@@ -809,7 +809,7 @@ M.funcs = {
       <
     ]=],
     name = 'bufname',
-    params = { { 'buf', 'any' } },
+    params = { { 'buf', 'number|string' } },
```
Comment (wookayin) [inline src/nvim/eval.lua:812]: number => integer?

**Review by wookayin — APPROVED** [review]
Nice improvements :)

**Comment by MariaSolOs** [comment]
> Nice improvements :)

Fixing the world one type annotation at a time ;)

**Pull request merged**